src/approaches/llm_probe_approach.py

```diff
@@ -0,0 +1,187 @@
+
+Example command line:
+    export OPENAI_API_KEY=<your API key>
+    python src/main.py --approach llm_probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_blocks_procedural_tasks \
+        --num_train_tasks 3 \
+        --num_test_tasks 1 \
+        --debug
+
+Easier setting:
+    python src/main.py --approach llm_probe --seed 0 \
+        --strips_learner oracle \
+        --env pddl_easy_delivery_procedural_tasks \
+        --pddl_easy_delivery_procedural_train_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_train_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_train_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_train_max_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_min_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_max_num_locs 2 \
+        --pddl_easy_delivery_procedural_test_min_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_max_want_locs 1 \
+        --pddl_easy_delivery_procedural_test_min_extra_newspapers 0 \
+        --pddl_easy_delivery_procedural_test_max_extra_newspapers 0 \
+        --num_train_tasks 5 \
+        --num_test_tasks 10 \
+        --debug
+"""
+from __future__ import annotations
+from typing import Dict, FrozenSet, Iterator, List, Optional, Set, Tuple
+from predicators.src.approaches import ApproachFailure
+from predicators.src import utils
+from predicators.src.approaches.open_loop_llm_approach import \
+    OpenLoopLLMApproach
+from predicators.src.settings import CFG
+from predicators.src.structs import Any, GroundAtom, ParameterizedOption, \
+    Sequence, State, _GroundNSRT
+
+
+class LLMProbeApproach(OpenLoopLLMApproach):
+    """LLMProbeApproach definition."""
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "llm_probe"
+    #
+    def _predict(self, state: State, atoms: Set[GroundAtom],
+                 goal: Set[GroundAtom], memory: Dict) -> _GroundNSRT:
+        # If we already have an abstract plan, execute the next step.
+        if "abstract_plan" in memory and memory["abstract_plan"]:
+            return memory["abstract_plan"].pop(0)
+        # Otherwise, we need to make a new abstract plan.
+        action_seq = self._run_planner(state, atoms, goal)
+
+        if action_seq is not None:
+            # If valid plan, add plan to memory so it can be refined!
+            memory["abstract_plan"] = action_seq
+            return memory["abstract_plan"].pop(0)
+        raise ApproachFailure("Approach failed to find solution")
+
+    def _run_planner(self, state: State, atoms: Set[GroundAtom],
+                     goal: Set[GroundAtom]) -> List[_GroundNSRT]:
+        init = atoms
+        initial_atoms: FrozenSet[GroundAtom] = frozenset(atoms)
+        objects = set(state)
+        trajectories = self._get_llm_based_state_action_predictions(
```

Comment:
this helper function helps a lot, thanks! let's keep going, and make two other helper functions: one for creating the policy from the trajectories, and one for setting up and running the policy-guided planning. then it will be very clear that this whole method is really doing three steps:
1. getting trajectories from the LLM
2. converting the trajectories into a policy
3. running policy-guided planning